### PR TITLE
Display token on Web OAuth2 success

### PR DESCRIPTION
**Pull request by s2lomon**
Thanks to this change, it's a bit easier to
use Web Oauth2 and composing it with --access-code
authentication in cli and jdbc.

### CODE REVIEW

**Comment by kokosing** [comment]
Fixes https://github.com/trinodb/trino/issues/6661

File: core/trino-main/src/main/resources/oauth2/success_ui.html [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    <link href="/ui/assets/presto.css" rel="stylesheet">
+</head>
+
+<body>
+
+<h2 class="text-center">OAuth2 authentication succeeded</h2>
+
+<h4 class="text-left">
+    <p>
+        <span>Go to <a href="/ui/">here</a> to access application.</span>
```
Comment (electrum) [inline core/trino-main/src/main/resources/oauth2/success_ui.html:56]:
```html
<span>Access the <a href="/ui/">Web Interface</a>.</span>
<span>The following JWT can be used in clients such as the CLI or JDBC driver:</span>

File: core/trino-main/src/main/resources/oauth2/success_ui.html [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        <span>Below is the jwt-token that can be used in cli or jdbc clients.</span>
+    </p>
+    <p>
+        <span>{OAUTH2_TOKEN}</span>
```
Comment (kokosing) [inline core/trino-main/src/main/resources/oauth2/success_ui.html:60]: It would be nice to display entire JDBC URL and Trino CLI command line so user can copy paste it all and just use. Also it would be nice to use some form that makes it easy to copy paste, like we are doing in docs.

File: core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Config.java [hunk 3]
```diff
@@ -211,4 +212,17 @@ public OAuth2Config setUserMappingFile(File userMappingFile)
         this.userMappingFile = Optional.ofNullable(userMappingFile);
         return this;
     }
+
+    public boolean isPrintOutTokenForWebUI()
+    {
+        return printOutTokenForWebUI;
+    }
+
+    @Config("web-ui.authentication.print-token")
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Config.java:221]: s/print/display?

web-ui.authentication.display-token.enabled?

File: core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Config.java [hunk 4]
```diff
@@ -211,4 +212,17 @@ public OAuth2Config setUserMappingFile(File userMappingFile)
...
+    @ConfigDescription("Regex to match against user name")
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Config.java:222]: ?

File: core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Config.java [hunk 5]
```diff
@@ -40,6 +40,7 @@
     private String jwksUrl;
     private String clientId;
     private String clientSecret;
+    private boolean printOutTokenForWebUI = true;
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Config.java:43]: please make it disabled by default

File: core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Service.java [hunk 6]
```diff
@@ -67,15 +68,17 @@
...
     @Inject
     public OAuth2Service(OAuth2Client client, @ForOAuth2 SigningKeyResolver signingKeyResolver, OAuth2Config oauth2Config)
             throws IOException
     {
         this.client = requireNonNull(client, "client is null");
         this.signingKeyResolver = requireNonNull(signingKeyResolver, "signingKeyResolver is null");
 
-        this.successHtml = Resources.toString(Resources.getResource(getClass(), "/oauth2/success.html"), UTF_8);
-        this.failureHtml = Resources.toString(Resources.getResource(getClass(), "/oauth2/failure.html"), UTF_8);
+        this.successHtml = Resources.toString(getResource(getClass(), "/oauth2/success.html"), UTF_8);
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Service.java:80]: static import should go to separate commit.

File: core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Service.java [hunk 7]
```diff
@@ -189,6 +193,18 @@ public String getSuccessHtml()
         return successHtml;
     }
 
+    public boolean showTokenOnSuccessfulWebLogin()
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Service.java:196]: isTokenDisplayEnabled()

File: core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Service.java [hunk 8]
```diff
@@ -189,6 +193,18 @@ public String getSuccessHtml()
...
+    {
+        return printOutTokenForWebUI;
+    }
+
+    public String getSuccessWebHtml(String token)
+            throws IOException
+    {
+        String successUITemplate = Resources.toString(getResource(getClass(), "/oauth2/success_ui.html"), UTF_8);
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Service.java:204]: Please move this to the constructor of `OAuth2UISuccessResource`

File: core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2ServiceModule.java [hunk 9]
```diff
@@ -40,6 +40,7 @@
     protected void setup(Binder binder)
     {
         jaxrsBinder(binder).bind(OAuth2CallbackResource.class);
+        jaxrsBinder(binder).bind(OAuth2UISuccessResource.class);
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2ServiceModule.java:43]: I think we should only load it when display is enabled

**Comment by HumairAK** [comment]
/poke 
any updates on this? would find this change very useful :)

**Comment by bitsondatadev** [comment]
:wave: @s2lomon - this PR is inactive and doesn't seem to be under development. If you'd like to continue work on this at any point in the future, feel free to re-open.
